src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java

```diff
@@ -819,29 +820,24 @@ private boolean validateAndTryUseToken(Logger logger, String token, SeaLightsPlu
         return true;
     }
 
-    private List<PomFile> getPomFiles(List<String> folders, String patterns, Logger logger, String pomPath) throws IOException, InterruptedException {
-        List<PomFile> pomFiles = new ArrayList<>();
-        boolean isParentPomInList = false;
+    private CollectedPomFiles getPomFiles(List<String> folders, String patterns, Logger logger, String pomPath) throws IOException, InterruptedException {
+        List<PomFile> nonRootPomFiles = new ArrayList<>();
         VirtualChannel channel = Computer.currentComputer().getChannel();
-        if (!patterns.startsWith("**" + File.separator))
+        if (!patterns.startsWith("**" + File.separator)) {
             patterns = "**" + File.separator + patterns;
+        }
 
         for (String folder : folders) {
             List<String> remotePoms = new FilePath(channel, folder).act(new SearchFileCallable(patterns));
             for (String matchingPom : remotePoms) {
                 logger.debug("Adding pom:" + matchingPom);
-                if (matchingPom.equalsIgnoreCase(pomPath))
-                    isParentPomInList = true;
-                pomFiles.add(new JenkinsPomFile(matchingPom, logger));
+                if (!matchingPom.equalsIgnoreCase(pomPath)) {
+                    nonRootPomFiles.add(new JenkinsPomFile(matchingPom, logger));
+                }
             }
         }
 
-
-        if (!isParentPomInList) {
-            pomFiles.add(new JenkinsPomFile(pomPath, logger));
-        }
-
-        return pomFiles;
+        return new CollectedPomFiles(new JenkinsPomFile(pomPath, logger), null, nonRootPomFiles);
```

Comment:
Why didnt you add `toList()` method and return a list? if so, you did not need to change the signature of the method

Reply:
because I need a `CollectedPomFiles` instance to pass to `MavenIntegrationInfo`'s constructor.